third_party/maya/lib/usdMaya/translatorMaterial.cpp

```diff
@@ -317,21 +317,11 @@ PxrUsdMayaTranslatorMaterial::ExportShadingEngines(
         return;
     }
 
-    if (PxrUsdMayaShadingModeExporter exporter =
+    if (auto exporterCreator =
             PxrUsdMayaShadingModeRegistry::GetExporter(shadingMode)) {
-        MItDependencyNodes shadingEngineIter(MFn::kShadingEngine);
-        for (; !shadingEngineIter.isDone(); shadingEngineIter.next()) {
-            MObject shadingEngine(shadingEngineIter.thisNode());
-            MFnDependencyNode seDepNode(shadingEngine);
-
-            PxrUsdMayaShadingModeExportContext c(
-                    shadingEngine,
-                    stage,
-                    mergeTransformAndShape,
-                    bindableRoots,
-                    overrideRootPath);
-
-            exporter(&c);
+        auto exporter = exporterCreator();
+        if (exporter != nullptr) {
```

Comment:
Since we used an inline assignment at line 320, can we make this if check also do the same for consistency? e.g.,

if (auto exporter = exporterCreator()) { ... }